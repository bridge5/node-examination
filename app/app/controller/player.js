'use strict';
const Controller = require('egg').Controller;

class PlayerController extends Controller {


    async test() {
        let params = {
            "name": "klkkkk",
            "playerId": 1,
            "position": "C"
        }
        let exsit = await this.ctx.service.player.findOne(params.playerId)
        if (!exsit) {
            exsit = await this.ctx.service.player.create(params)
        }
        console.log(exsit, '---数据库结果')

        params.name = 'updateKkkkk'
        params.position = 'B'
        let updateRes = await this.ctx.service.player.update(params)

        console.log(updateRes, '---修改结果')

        let delRes = await this.ctx.service.player.del(params.playerId)

        console.log(delRes, '---删除结果')
    }

    async create() {
        try {
            let params = this.ctx.form.create.checkData({
                ...this.ctx.request.body
            })
            let res = await this.ctx.service.player.create(params)
            return this.ctx.body = this.ctx.out.success(res)
        } catch (err) {
            return this.ctx.body = this.ctx.out.error(err.code, err.message);
        }
    }

    async update() {
        try {
            let params = this.ctx.form.create.checkData({
                ...this.ctx.request.body
            })
            let res = await this.ctx.service.player.update(params)
            return this.ctx.body = this.ctx.out.success(res)
        } catch (err) {
            return this.ctx.body = this.ctx.out.error(err.code, err.message);
        }
    }

    async findOne() {
        try {
            let urlArr = this.ctx.request.url.split('/')
            let params = this.ctx.form.findOne.checkData({
                ...this.ctx.request.query,
                playerId: parseInt(urlArr[urlArr.length - 1])
            })
            let res = await this.ctx.service.player.findOne(params.playerId)
            return this.ctx.body = this.ctx.out.success(res)
        } catch (err) {
            return this.ctx.body = this.ctx.out.error(err.code, err.message);
        }
    }


    async del() {
        try {
            let urlArr = this.ctx.request.url.split('/')
            let params = this.ctx.form.findOne.checkData({
                ...this.ctx.request.query,
                playerId: parseInt(urlArr[urlArr.length - 1])
            })
            let res = await this.ctx.service.player.del(params.playerId)
            return this.ctx.body = this.ctx.out.success(res)
        } catch (err) {
            return this.ctx.body = this.ctx.out.error(err.code, err.message);
        }
    }

}

module.exports = PlayerController;