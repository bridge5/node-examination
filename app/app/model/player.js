/* indent size: 2 */

module.exports = function(sequelize, DataTypes) {
  const Player = sequelize.define('Player', {
    id: {
      type: DataTypes.INTEGER(11),
      allowNull: false,
      primaryKey: true,
      autoIncrement: true,
    },
    playerId: {
      type: DataTypes.INTEGER(11),
      allowNull: false,
      defaultValue: '0',
      field: 'player_id',
    },
    name: {
      type: DataTypes.STRING(50),
      allowNull: false,
      defaultValue: '',
    },
    position: {
      type: DataTypes.STRING(50),
      allowNull: false,
      defaultValue: '',
    },
    created: {
      type: DataTypes.INTEGER(11),
      allowNull: false,
      defaultValue: '0',
    },
    modified: {
      type: DataTypes.INTEGER(11),
      allowNull: false,
      defaultValue: '0',
    },
  }, {
    freezeTableName: true,
    timestamps: false,
    tableName: 'player',
  });


  return Player;
};
