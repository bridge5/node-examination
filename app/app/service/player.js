'use strict';
const BaseService = require('./base_model_define');
class PlayerService extends BaseService {
  constructor(ctx) {
    super(ctx);
  }

  async create(data) {
    const exsit = await this.Player.findOne({
      where: {
        playerId: data.playerId,
      },
      raw: true,
    });
    if (exsit) {
      return exsit;
    }

    const tn = this.ctx.helper.currentTimestamp();
    data.created = tn;
    data.modified = tn;
    return await this.Player.create(data);
  }

  async update(data) {
    const tn = this.ctx.helper.currentTimestamp();
    data.created = tn;
    data.modified = tn;
    return await this.Player.update(data, {
      where: {
        playerId: data.playerId,
      },
    });
  }


  async findOne(playerId) {
    return await this.Player.findOne({
      where: {
        playerId,
      },
      raw: true,
    });
  }

  async del(playerId) {
    return await this.Player.destroy({
      where: {
        playerId,
      },
    });
  }


}

module.exports = PlayerService;
