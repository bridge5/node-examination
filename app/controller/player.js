/**
 * File: player.js
 * Project: node-examination
 * FilePath: /controller/player.js
 * Created Date: 2021-05-24 18:22:05
 * Author: SaltFish l1218838196@gmail.com
 * -----
 * Last Modified: 2021-05-25 16:29:55
 * Modified By: SaltFish
 * -----
 * Description:
 */

const Router = require('koa-router');
const name = 'Player';

class PlayerController {
  constructor() {
    this.name = 'Player';
    this.router = new Router({
      prefix: '/player'
    });
    this.initRouters()
  }

  initRouters = () => {
    this.router.post('/', this.create)
    this.router.put('/:playerId', this.updateById)
    this.router.get('/:playerId', this.findById)
    this.router.delete('/:playerId', this.deleteById)
  }
  /**
   * 
   * @param {Router.IRouterContext} ctx 
   * @param {string[]} requiredParams
   * @param {object} values
   */
  validator = (ctx, requiredParams = [], values = {}) => {
    const resouceValidator = ctx.validators[this.name];
    if (ctx.params[`${this.name.toLowerCase()}Id`]) {
      try {
        resouceValidator['id'](parseInt(ctx.params.playerId))
      } catch (error) {
        ctx.body = error.message
        ctx.status = 400;
        return false
      }
    }

    // 检查必传参数
    for (let i = 0; i < requiredParams.length; i++) {
      const paramName = requiredParams[i];
      if (typeof values[paramName] === 'undefined') {
        ctx.body = `${paramName} is required!`
        ctx.status = 405;
        return false
      }
    }

    for (const key in values) {
      if (Object.hasOwnProperty.call(values, key)) {
        if (!Object.hasOwnProperty.call(resouceValidator, key)) {
          ctx.body = `${key} is not in schema`
          ctx.status = 405
          return false
        }
        const element = values[key];
        try {
          if (element) {
            resouceValidator[key](element)
          }
        } catch (error) {
          console.log(error)
          ctx.body = `error params '${key}': ${error.message}`
          ctx.status = 405
          return false
        }
      }
    }

    return true
  }

  dbDataTrans = (dbData) => {
    if (dbData.toObject) {
      dbData = dbData.toObject()
    }
    const { _id } = dbData;
    delete dbData._id
    delete dbData.__v
    dbData.id = _id
    return dbData
  }
  /**
   * 
   * @param {Router.IRouterContext} ctx 
   * @param {number} id 
   */
  existBrforeAction = async (ctx, id) => {
    const model = ctx.models[this.name];
    const exist = await model.findById(id).exec();
    if (exist) {
      return exist
    }
    return null
  }

  /**
   * 
   * @param {Router.IRouterContext} ctx 
   */
  create = async (ctx) => {
    const body = ctx.request.body
    const model = ctx.models[this.name]
    const passVerification = this.validator(ctx, ['name'], {
      ...body
    })

    if (!passVerification) {
      return
    }
    try {
      if (body.id) {
        const exist = await this.existBrforeAction(ctx, body.id)

        if (exist) {
          ctx.status = 412;
          ctx.body = `Id ${body.id} already exist!`
          return;
        }
        body._id = body.id
      }
      const result = await model.create({
        ...body
      })
      ctx.status = 200;
      ctx.body = this.dbDataTrans(result)
    } catch (error) {
      console.error(error)
      ctx.throw(error)
    }
  }

  /**
   * 
   * @param {Router.IRouterContext} ctx 
   */
  findById = async (ctx) => {
    const passVerification = this.validator(ctx);

    if (!passVerification) {
      return;
    };
    const { playerId } = ctx.params
    const id = parseInt(playerId)
    try {
      const exist = await this.existBrforeAction(ctx, id)
      if (!exist) {
        ctx.status = 404;
        ctx.body = `Id ${id} not exist!`
        return;
      }
      ctx.status = 200;
      ctx.body = this.dbDataTrans(exist)
    } catch (error) {
      console.error(error)
      ctx.throw(error)
    }
  }

  /**
  * 
  * @param {Router.IRouterContext} ctx 
  */
  updateById = async (ctx) => {
    const passVerification = this.validator(ctx, [], {
      ...ctx.request.body
    });

    if (!passVerification) {
      return;
    };
    const { playerId } = ctx.params
    const id = parseInt(playerId)
    try {
      const exist = await this.existBrforeAction(ctx, id)
      if (!exist) {
        ctx.status = 404;
        ctx.body = `Id ${id} not exist!`
        return;
      }
      await exist.set({
        ...ctx.request.body
      }).save()
      ctx.status = 200;
      ctx.body = this.dbDataTrans(exist)
    } catch (error) {
      console.error(error)
      ctx.throw(error)
    }
  }

  /**
   * 
   * @param {Router.IRouterContext} ctx 
   */
  deleteById = async (ctx) => {
    const passVerification = this.validator(ctx);
    if (!passVerification) {
      return;
    };
    const { playerId } = ctx.params
    const id = parseInt(playerId)
    try {
      const exist = await this.existBrforeAction(ctx, id)
      if (!exist) {
        ctx.status = 404;
        ctx.body = `Id ${id} not exist!`
        return;
      }
      await exist.remove()
      ctx.status = 200;
      ctx.body = this.dbDataTrans(exist)
    } catch (error) {
      console.error(error)
      ctx.throw(error)
    }
  }
}

module.exports = PlayerController