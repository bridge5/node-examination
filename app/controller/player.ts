import { mysqlConnect } from "../common/database/mysql"
import { httpCode } from "../common/httpCode/httpCode";

// Create a new player
export const createPlayer = async (req, res) => {
    const { name, position } = req.body;
    if (name == undefined || position == undefined) {
        res.status(405).json(httpCode.INVALIDINPUT);
        return
    }
    const sql = `insert into player (name, position) values ("${name}", "${position}")`;
    const data = await mysqlConnect(sql);
    const { insertId } = data;
    const player = {
        id: insertId,
        name,
        position
    }
    res.json(player);
}

// Update an existing player
export const updatePlayer = async (req, res) => {
    const { playerId, name, position } = req.body;
    if (playerId == undefined) {
        res.status(405).json(httpCode.INVALID);
        return
    }
    if (name == undefined || position == undefined) {
        res.status(405).json(httpCode.INVALIDINPUT);
        return
    }
    const sql = `update player set name = "${name}", position = "${position}" where id = "${playerId}"`;
    const data = await mysqlConnect(sql);
    const { affectedRows } = data
    if (affectedRows == 0) {
        res.status(404).json(httpCode.NOTFOUND);
        return
    }
    const player = {
        id: playerId,
        name,
        position
    }
    res.json(player);
}

// Find player by ID
export const getPlayer = async (req, res) => {
    const { playerId } = req.params;
    if (playerId == undefined) {
        res.status(405).json(httpCode.INVALID);
        return
    }
    const sql = `select * from player where id = ${playerId}`;
    const data = await mysqlConnect(sql);
    if (data.length == 0 ) {
        res.status(404).json(httpCode.NOTFOUND);
        return
    }

    res.json(data[0]);
}

export const deletePlayer = async (req, res) => {
    const { playerId } = req.params;
    if (playerId == undefined) {
        res.status(405).json(httpCode.INVALID);
        return
    }
    const sql = `delete from player where id = ${playerId}`;
    const data = await mysqlConnect(sql);
    const { affectedRows } = data
    if (affectedRows == 0) {
        res.status(404).json(httpCode.NOTFOUND);
        return
    }
    res.json({
        playerId
    })
}

