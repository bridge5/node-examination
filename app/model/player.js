'use strict';

const moment = require('moment');

const {Sequelize, pool} = require('../common/mysql');
const C = require('../common/constant');

const Player = pool.define('bas_player', {
  // 用户 player id
  playerId: {
    type: Sequelize.INTEGER,
    field: 'player_id',
    autoIncrement: true,
    primaryKey: true,
  },
  // 用户名称
  name: {
    type: Sequelize.STRING(32),
    field: 'name',
    unique: true,
  },
  // 年龄
  age: {
    type: Sequelize.INTEGER,
    field: 'age',
    unique: true,
  },
  // 手机号码
  phone: {
    type: Sequelize.STRING(32),
    field: 'phone',
    unique: true,
  },
  // 账号创建时间
  ctime: {
    type: Sequelize.DATE,
    field: 'ctime',
    defaultValue: moment().format('YYYY-MM-DD HH:mm:ss'),
  },
  // 修改时间
  mtime: {
    type: Sequelize.DATE,
    field: 'mtime',
    defaultValue: moment().format('YYYY-MM-DD HH:mm:ss'),
  },
  // 是否有效
  invaild: {
    type: Sequelize.CHAR(1),
    field: 'invaild',
    defaultValue: C.CHAR.NO,
  },
}, {
  // 启用时间
  timestamps: false,
  // 表注释
  comment: '用户基本信息表',
  freezeTableName: true,
});


/**
 * 创建用户数据
 * @param {Object} data
 */
exports.addPlayer = async (data) => {
  const res = await Player.create(
    data,
  );

  return res;
};


/**
 * 更新用户信息
 * @param {Object} where
 * @returns {Promise<[number, any[]]}
 */
exports.updatePlayer = async (data, where) => {
  const res = await Player.update(
    data,
    {
      where,
    },
  );

  return res;
};


/**
 * 获取用户信息
 * @param {Object} where
 */
exports.getPlayer = async (where) => {
  const res = await Player.findOne({
    where,
  });
  return res;
};