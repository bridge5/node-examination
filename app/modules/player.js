'use strict';
const mongodber = require('../lib/mongodber');
const mongoPlayer = mongodber._db.player;

// 创建球员
async function postCreatePlayer(requestData){
    const  returnJson = {};

    const playInfo = await mongoPlayer.collection('Player').findOne({ id: requestData.id });
    if (playInfo) throw Error('Invalid input');

    await mongoPlayer.collection('Player').insertOne(requestData);

    return returnJson;
}

// 修改球员
async function putUpdatePlayer(requestData){
    const  returnJson = {};

    if (!requestData.name && !requestData.position){
        throw new Error('Validation exception');
    }

    const playInfo = await mongoPlayer.collection('Player').findOne({ id: requestData.id });
    if (!playInfo) throw Error('Player not found');

    const updateInfo = {};
    if (requestData.name) updateInfo.name = requestData.name;
    if (requestData.position) updateInfo.position = requestData.position;

    await mongoPlayer.collection('Player').updateOne({ id: requestData.id }, { $set: updateInfo });

    return returnJson;
}

// 获取球员
async function getPlayer(playId){
    const playInfo = await mongoPlayer.collection('Player').findOne({ id: playId });
    if (!playInfo) throw Error('Player not found');

    const  returnJson = {
        name: playInfo.name,
        id: playInfo.id,
        position: playInfo.position,
    };

    return returnJson;
}

// 删除球员
async function deletePlayer(playId){
    const returnJson = {};

    const playInfo = await mongoPlayer.collection('Player').findOne({ id: playId });
    if (!playInfo) throw Error('Player not found');

    await mongoPlayer.collection('Player').deleteOne({ id: playId });

    return returnJson;
}

module.exports = {
    postCreatePlayer,
    putUpdatePlayer,
    getPlayer,
    deletePlayer,
};