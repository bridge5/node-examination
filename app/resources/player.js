/**
 * File: player.js
 * Project: node-examination
 * FilePath: /resources/player.js
 * Created Date: 2021-05-24 16:21:29
 * Author: SaltFish l1218838196@gmail.com
 * -----
 * Last Modified: 2021-05-24 23:28:44
 * Modified By: SaltFish
 * -----
 * Description:
 */
const autoIncrement = require('mongoose-auto-increment');
const { Schema } = require('mongoose')

const postionEnum = [
  'C', 'PF', 'SF', 'PG', 'SG'
]

module.exports = {
  name: "Player",
  schema: new Schema({
    // id: {
    //   required: true,
    //   type: Schema.Types.Number
    // },
    name: {
      required: true,
      type: Schema.Types.String
    },
    position: {
      type: Schema.Types.String
    },
  }),
  validator: {
    id: (val) => {
      if (typeof val !== 'number' || !Number.isInteger(val)) {
        throw new TypeError('id must be integer')
      }
    },
    name: (val) => {
      if (typeof val !== 'string') {
        throw new TypeError('name must be string')
      }
    },
    position: (val) => {
      if (typeof val !== 'string') {
        throw new TypeError('position must be string')
      }
      if (postionEnum.indexOf(val) === -1) {
        throw new TypeError(`position must valueof [${postionEnum.join(',')}]`)
      }
    },
  }
}