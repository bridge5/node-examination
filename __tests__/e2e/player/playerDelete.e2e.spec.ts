import { HttpStatus, INestApplication } from '@nestjs/common';
import request from 'supertest';
import * as mockingoose from 'mockingoose';
import {
  generateSequentialTests,
  mockGenericApp,
  playerTestModel,
} from '../../helper/e2e.helper';
import { PlayerModule } from '../../../src/modules/player/player.module';
import { PlayerHandlerService } from '../../../src/modules/player/services/player.handler.service';
import { playerValidExample } from '../../../src/modules/player/schema/player.schema';
import { UpdatePlayerService } from '../../../src/modules/player/services/updatePlayer.service';
import { QueryPlayerService } from '../../../src/modules/player/services/queryPlayer.service';
import { playerIdParamValidExample } from '../../../src/dto/playerIdParam.dto';

const path = `/player/${playerIdParamValidExample.playerId}`;

describe(`DELETE ${path}`, () => {
  let app: INestApplication;
  let playerHandlerService;
  let queryPlayerService: QueryPlayerService;
  let updatePlayerService: UpdatePlayerService;
  let mockFunctionGroup;

  beforeAll(async () => {
    const appModule = (
      await mockGenericApp({
        module: {
          imports: [PlayerModule],
        },
      })
    ).createNestApplication();
    playerHandlerService = appModule.get<PlayerHandlerService>(
      'PlayerHandlerService',
    );
    queryPlayerService =
      appModule.get<QueryPlayerService>('QueryPlayerService');
    updatePlayerService = appModule.get<UpdatePlayerService>(
      'UpdatePlayerService',
    );
    mockFunctionGroup = {
      spyHandlerDeletePlayerById: jest
        .spyOn(playerHandlerService, 'deletePlayerById')
        .mockName('spyHandlerDeletePlayerById'),
      spyFindOne: jest
        .spyOn(queryPlayerService, 'findOne')
        .mockName('spyFindOne'),
      spyUpdate: jest
        .spyOn(updatePlayerService, 'update')
        .mockName('spyUpdate'),
    };
    app = await appModule.init();
  });

  describe('success cases', () => {
    it('should delete player', async () => {
      mockingoose(playerTestModel).toReturn(playerValidExample, 'findOne');
      mockingoose(playerTestModel).toReturn(null, 'findOneAndUpdate');
      const response = await request(app.getHttpServer()).delete(path);
      expect(response.status).toEqual(HttpStatus.OK);
      generateSequentialTests({
        mockFunctionGroup,
        apiResponse: response,
        snapshots: ['spyHandlerDeletePlayerById', 'spyFindOne', 'spyUpdate'],
      });
    });
  });

  describe('fail cases', () => {
    it('should throw Validation Error if player id is incorrect', async () => {
      const response = await request(app.getHttpServer()).delete('/player/aaa');
      expect(response.status).toEqual(405);
      generateSequentialTests({
        mockFunctionGroup,
        apiResponse: response,
        notCalled: ['spyHandlerDeletePlayerById', 'spyFindOne', 'spyUpdate'],
      });
    });
    it('should throw NotFoundErrorException if player is not fount', async () => {
      mockingoose(playerTestModel).toReturn(null, 'findOne');
      const response = await request(app.getHttpServer()).delete(path);
      expect(response.status).toEqual(HttpStatus.NOT_FOUND);
      generateSequentialTests({
        mockFunctionGroup,
        apiResponse: response,
        snapshots: ['spyHandlerDeletePlayerById', 'spyFindOne'],
        notCalled: ['spyUpdate'],
      });
    });
  });

  afterEach(() => {
    mockingoose.resetAll();
    jest.clearAllMocks();
  });

  afterAll(async () => {
    await app.close();
  });
});
