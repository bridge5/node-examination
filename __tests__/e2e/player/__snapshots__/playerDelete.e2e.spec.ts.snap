// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DELETE /player/5b8ea08eb0b2b900dcb8be53 fail cases should throw NotFoundErrorException if player is not fount 1`] = `
[MockFunction spyHandlerDeletePlayerById] {
  "calls": Array [
    Array [
      "5b8ea08eb0b2b900dcb8be53",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`DELETE /player/5b8ea08eb0b2b900dcb8be53 fail cases should throw NotFoundErrorException if player is not fount 2`] = `
[MockFunction spyFindOne] {
  "calls": Array [
    Array [
      Object {
        "filter": Object {
          "_id": "5b8ea08eb0b2b900dcb8be53",
          "status": 1,
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`DELETE /player/5b8ea08eb0b2b900dcb8be53 fail cases should throw NotFoundErrorException if player is not fount 3`] = `
Object {
  "code": 404,
  "message": "player not fount",
  "path": "/player/5b8ea08eb0b2b900dcb8be53",
  "statusCode": 404,
  "type": "NotFoundErrorException",
}
`;

exports[`DELETE /player/5b8ea08eb0b2b900dcb8be53 fail cases should throw Validation Error if player id is incorrect 1`] = `
Object {
  "code": 405,
  "message": Array [
    Object {
      "msg": "{\\"isMongoId\\":\\"playerId must be a mongodb id\\"}",
      "path": "playerId",
      "value": "aaa",
    },
  ],
  "path": "/player/aaa",
  "statusCode": 405,
  "type": "ValidationErrorException",
}
`;

exports[`DELETE /player/5b8ea08eb0b2b900dcb8be53 success cases should delete player 1`] = `
[MockFunction spyHandlerDeletePlayerById] {
  "calls": Array [
    Array [
      "5b8ea08eb0b2b900dcb8be53",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`DELETE /player/5b8ea08eb0b2b900dcb8be53 success cases should delete player 2`] = `
[MockFunction spyFindOne] {
  "calls": Array [
    Array [
      Object {
        "filter": Object {
          "_id": "5b8ea08eb0b2b900dcb8be53",
          "status": 1,
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`DELETE /player/5b8ea08eb0b2b900dcb8be53 success cases should delete player 3`] = `
[MockFunction spyUpdate] {
  "calls": Array [
    Array [
      Object {
        "conditions": Object {
          "_id": "5b8ea08eb0b2b900dcb8be53",
        },
        "update": Object {
          "status": 0,
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`DELETE /player/5b8ea08eb0b2b900dcb8be53 success cases should delete player 4`] = `Object {}`;
